Doesn't compile on 32bit

This function created a compile error when trying to build with arm32 (```CGO_ENABLED=0 GOOS=linux GOARCH=arm go build test.go```)
https://github.com/puzpuzpuz/xsync/blob/d5f0f9d5b79c39bf68bf9771f52f477262972f6f/util.go#L18-L24

Following error is shown: 
```
#0 7.331 /go/pkg/mod/github.com/puzpuzpuz/xsync@v1.2.1/util.go:20:24: 0xff51afd7ed558ccd (untyped int constant 18397679294719823053) overflows uintptr
#0 7.331 /go/pkg/mod/github.com/puzpuzpuz/xsync@v1.2.1/util.go:21:24: 0xc4ceb9fe1a85ec53 (untyped int constant 14181476777654086739) overflows uintptr
```

This can likely be fixed by casting the argument to uint64 before performing the operations.

--- a/map.go
+++ b/map.go
@@ -49,12 +49,12 @@ const (
 // are supported. That's because Golang standard library does not
 // expose the built-in hash functions for interface{} values.
 type Map struct {
-	table        unsafe.Pointer // *mapTable
+	totalGrowths int64
+	totalShrinks int64
 	resizing     int64          // resize in progress flag; updated atomically
 	resizeMu     sync.Mutex     // only used along with resizeCond
 	resizeCond   sync.Cond      // used to wake up resize waiters (concurrent modifications)
-	totalGrowths int64
-	totalShrinks int64
+	table        unsafe.Pointer // *mapTable
 }
 
 type mapTable struct {

--- a/mapof.go
+++ b/mapof.go
@@ -31,12 +31,12 @@ import (
 // are supported. That's because Golang standard library does not
 // expose the built-in hash functions for interface{} values.
 type MapOf[V any] struct {
-	table        unsafe.Pointer // *mapTable
+	totalGrowths int64
+	totalShrinks int64
 	resizing     int64          // resize in progress flag; updated atomically
 	resizeMu     sync.Mutex     // only used along with resizeCond
 	resizeCond   sync.Cond      // used to wake up resize waiters (concurrent modifications)
-	totalGrowths int64
-	totalShrinks int64
+	table        unsafe.Pointer // *mapTable
 }
 
 // NewMapOf creates a new MapOf instance.

--- a/util.go
+++ b/util.go
@@ -16,11 +16,12 @@ const (
 )
 
 // murmurhash3 64-bit finalizer
-func hash64(x uintptr) uint64 {
+func hash64(v uintptr) uint64 {
+	x := uint64(v)
 	x = ((x >> 33) ^ x) * 0xff51afd7ed558ccd
 	x = ((x >> 33) ^ x) * 0xc4ceb9fe1a85ec53
 	x = (x >> 33) ^ x
-	return uint64(x)
+	return x
 }
 
 // exposes the built-in memhash function

